src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.TestService/TestService.cs

```diff
@@ -38,7 +38,7 @@ public TestService(string serviceName, Exception throwException = null)
             this._exception = throwException;
 
             this._serverStream = new NamedPipeServerStream(serviceName);
-            _waitClientConnect = this._serverStream.WaitForConnectionAsync();
+            this._serverStream.WaitForConnectionAsync().ContinueWith((t) => WriteStreamAsync(PipeMessageByteCode.Connected).Wait());
```

Comment:
We should not use Wait to block like this.  Why do we need to do this connection in the TestService constructor?  It would be better to either a) have a static factory async Task method that can construct one of these and then do the asynchronous work, or b) move this waiting to whatever test actually needs this.

Reply:
We are doing it in a constructor because once the service has started, we have no control over it. Its a completely different process. We are starting this service using Interop and Sc Manager.
We can remove the last  ```.wait()``` as we are already waiting in the tests that are using this namedPipe. How is WaitForconnectionAsync() blocking ?